StoreKitUnitTests/TrialOrIntroPriceEligibilityCheckerSK2Tests.swift

```diff
@@ -131,4 +131,122 @@ class TrialOrIntroPriceEligibilityCheckerSK2Tests: StoreKitConfigTestCase {
             expect(receivedEligibility.status) == expected[product]
         }
     }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testCheckEligibilityForProductIsEligibleForEligibleSubscription() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let productIdentifiers = Set([
+            "com.revenuecat.monthly_4.99.1_week_intro"
+        ])
+
+        let sk2Product = try await ProductsFetcherSK2().products(identifiers: productIdentifiers).first
+        let receivedProduct = try XCTUnwrap(sk2Product)
+        let storeProduct = StoreProduct.from(product: receivedProduct)
+
+        var completionCalled = false
+
+        let status: IntroEligibilityStatus = try await withCheckedThrowingContinuation({ continuation in
+            self.trialOrIntroPriceEligibilityChecker.checkEligibility(product: storeProduct) { status in
+                completionCalled = true
+                continuation.resume(returning: status)
+            }
+        })
+
+        expect(completionCalled) == true
+        expect(status) == .eligible
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testCheckEligibilityForLifetimeProductIsNoIntroOfferExists() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let productIdentifiers = Set([
+            "lifetime"
+        ])
+
+        let sk2Product = try await ProductsFetcherSK2().products(identifiers: productIdentifiers).first
+        let receivedProduct = try XCTUnwrap(sk2Product)
+        let storeProduct = StoreProduct.from(product: receivedProduct)
+
+        var completionCalled = false
+
+        let status: IntroEligibilityStatus = try await withCheckedThrowingContinuation({ continuation in
```

Comment:
I wonder if we should make `TrialOrIntroPriceEligibilityChecker` `async` by default since that's "the future", which would remove the need to do this dance here. But we can do that in a follow up PR.

Reply:
wouldn't that make it incompatible with iOS 11-12? If you have a completion-block alternative that's calling async / await internally, you won't be able to use it for those OS verisons

Reply:
Oh that's right never mind!